src/serializer/visitors.js

```diff
@@ -56,6 +56,27 @@ function replaceName(path, residualFunctionBinding, name, data) {
   }
 }
 
+function getLiteralTruthiness(node) {
+  // Returns { known: boolean, value?: boolean }. 'known' is true only if this is a literal of known truthiness and with no side effects; if 'known' is true, 'value' is its truthiness.
+  if (t.isBooleanLiteral(node) || t.isNumericLiteral(node) || t.isStringLiteral(node)) {
+    return { known: true, value: !!node.value };
+  }
+  if (
+    t.isFunctionExpression(node) ||
+    t.isArrowFunctionExpression(node) ||
+    t.isRegExpLiteral(node) ||
+    (t.isClassExpression(node) && node.superClass === null) ||
+    (t.isObjectExpression(node) && node.properties.length === 0) ||
+    (t.isArrayExpression(node) && node.elements.length === 0)
+  ) {
+    return { known: true, value: true };
+  }
+  if (t.isNullLiteral(node)) {
+    return { known: true, value: false };
+  }
+  return { known: false };
```

Comment:
Another case you could add: If the node is a unary expression with operator `void` and the argument is a literal, then this node will evaluate to `undefined` without side effects, and thus results in the known truthiness value of `false`.

Reply:
This is true. I could also handle `!false`, `+0`, `[0, true]`, `void void [ +!'a', { foo: 0, get bar(){ return whatever(); } } ]`, `typeof /a/`, and so on. But one could go a long, long way down this road, and I think it probably makes sense to do the remainder of that work as part of partial evaluation, rather than in an ad-hoc way here.